build.md

```diff
@@ -0,0 +1,145 @@
+Congratulations!
+
+The built `vphysics_jolt.dll` file should now be located in `mini-source-sdk/sdk2013-mp/game/bin/`.
+
+### Build errors
+
+Common build errors you may run into building on windows and their solutions.
+
+> #### memoverride.cpp
+
+If there is an error about involving a symbol named ` size_t _msize` or `size_t __cdecl _msize_base` try removing or commenting out definitions of <strong>both</strong>. You should leave only the code below:
+```cpp
+size_t msize( void *pMem )
+{
+	return g_pMemAlloc->GetSize(pMem);
+}
+```
+
+Where the code to be removed/commented would be:
+```cpp
+size_t __cdecl _msize_base( void *pMem ) noexcept
+{
+	return g_pMemAlloc->GetSize(pMem);
+}
+
+size_t _msize( void *pMem )
+{
+	return _msize_base(pMem);
+}
+```
+
+###### It may be wise to try different variations if the given solution does not work.
+
+## Linux
+
+Continuing on linux, there are a few prerequisites to building regardless of distrobution.
+- glibc v2.29+
+- gcc v8+
+- g++ v8+
+- gcc-multilib
+- g++-multilib
+- make
+
+###### Note: Attempts to compile with older versions of glibc are undocumented and probably will not work (you are still welcome to confirm this)
+
+### Debian/Ubuntu
+###### Note: Some older versions of debian or debian-based distrobutions (i.e. ubuntu) may use `apt-get`
+###### Note: Older versions may also require version explicit packages if available.
+
+Before we continue building project files, lets get our dependencies out of the way.
+Install the `build-essential` group to get common developer packages such as gcc, g++, and make, which are included.
+```bash
+sudo apt install build-essential
+```
+
+After installing the build-essential group packages, we now need to install the x86 dependencies.
+```bash
+sudo apt install gcc-multilib g++-multilib
+```
+
+The next step is to build our makefiles by running the premade script `createjoltprojects.sh`
+```bash
+./createjoltprojects.sh
+```
+
+Once the script has finished running you should now have a file name `jolt.mak` in the current directory. Verify you do and now run it with the command below. This command automatically will (and is recommended to) use all available CPU threads. If you want to specify it yourself, replace `` `nproc` `` with a number less than or equal to your max number of threads. To be clear, this only affects compile speed.
+```bash
+make -j `nproc` -f jolt.mak
+```
+
+Alternative example...
+```bash
+make -j 8 -f jolt.mak
+```
+
+Congratulations!
+
+If the console outputs `LINKING` *and* `COPYING TO` messages, that means your build was successful and you may retrieve your `vphysics_jolt.so` and `vphysics_jolt_srv.so` files which will be located in `mini-source-sdk/sdk2013-mp/game/bin/`.
+
+### Build errors (distrobution independent)
+
+Common build errors you may run into building on linux and their solutions.
+
+> #### error: unrecognized command line option ‘-std=gnu++20’; did you mean ‘-std=gnu++2a’? (make: *** [jolt.mak:28: all] Error 2)
```

Comment:
No longer needed if we just require GCC 10+

Reply:
I believe I was on 9, I'll try to get it installed and rebuild this too